ExternalModule.php

```diff
@@ -362,4 +362,59 @@ function sendEmail($email_info) {
 		return $success;
     }
 
+    function addEmailAlertsToAllProjectComments() {
+        $projects = $this->framework->getProjectsWithModuleEnabled();
+        foreach($projects as $project_id) {
+            $this->addEmailAlertsToProjectComments($project_id);
+        }
+    }
+
+    function addEmailAlertsToProjectComments($project_id) {
+        $sql = "SELECT record, event_id, last_sent FROM (
+                (SELECT * FROM `redcap_alerts`
+                    WHERE
+                    project_id = $project_id
+                 AND alert_title = 'Negative result swab'
+                 ) as ra
+                INNER JOIN redcap_alerts_sent AS ras ON ras.alert_id = ra.alert_id
+            );";
+        $result = $this->framework->query($sql);
+        $results = $result->fetch_all(MYSQLI_ASSOC);
+
+        if (empty($results)) return;
+
+        $records = [];
+        $events = [];
+        foreach($results as $result) {
+            array_push($records, $result['record']);
+            array_push($events, $result['event_id']);
+        }
+
+
+        $get_data = [
+            'project_id' => $project_id,
+            'records' => array_values($records),
+            'events' => array_values($events)
+            ];
+
+        $redcap_data = \REDCap::getData($get_data);
+
+        foreach ($results as $result) {
+            $comment_contents = "Negative alert email sent: ";
```

Comment:
I want to plan for the next message, mostly likely, the antibody result message. What if we were to use the `alert_title` to build up the text we write into the result_comments field?  e.g. 

```suggestion
            $comment_contents = alert_title . " email sent: ";
```

This would put @tlstoffs' alert titles into the forefront. She might want to review them for readability in this context.

Reply:
I agree, I thought that would be v2 of this feature as we did it since we were still up in the air on what is getting email alerts. I could add `alert_title` as a column in the SQL and make it a variable. Let's discuss with @tlstoffs.

Reply:
This one is more important than the second one ATM.  The second one depends on us have stable strings in the comments field.  unless you just want to check the time stamp; it might be enough.